shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_MACOS_FRAMEWORK_SOURCE_FLUTTERPLATFORMNODEDELEGATEMAC_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_MACOS_FRAMEWORK_SOURCE_FLUTTERPLATFORMNODEDELEGATEMAC_H_
+
+#import <Cocoa/Cocoa.h>
+
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterEngine.h"
+
+#include "flutter/shell/platform/common/cpp/flutter_platform_node_delegate.h"
+#include "flutter/shell/platform/embedder/embedder.h"
+
+static constexpr int32_t kRootNode = 0;
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// The macOS implementation of FlutterPlatformNodeDelegate. This class uses
+/// AXPlaformNodeMac to create native accessibility object.
+class FlutterPlatformNodeDelegateMac : public FlutterPlatformNodeDelegate {
+ public:
+  FlutterPlatformNodeDelegateMac(__weak FlutterEngine* engine);
+  ~FlutterPlatformNodeDelegateMac();
```

Comment:
there is no more virtual method on this class, do we still make the destructor virtual? or is this part of the style guide?

Reply:
It's inheriting from a superclass, so presumably there are references to instances of this class that are typed as the superclass. If that happens to an owning reference, and all the classes involved don't have virtual destructors, then the object won't be destroyed correctly. It's not a style issue, it's a fundamental correctness issue.

Reply:
I am ok making this virtual. However, i am a bit confused according to this post (the second answer) https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors

the destructor is only skipped if you inherited from a class that has non virtual destructor? If that is the case, this will only be a problem if another class inherited FlutterPlatformNodeDelegateMac and delete the pointer while typed it as FlutterPlatformNodeDelegateMac. Am i missing something?

Reply:
> the destructor is only skipped if you inherited from a class that has non virtual destructor? If that is the case, this will only be a problem if another class inherited FlutterPlatformNodeDelegateMac and delete the pointer while typed it as FlutterPlatformNodeDelegateMac. Am i missing something?

Sorry, yes; I misremembered the technical specifics here. I thought the whole chain had to be declared `virtual` explicitly. I would still mark it as `virtual` since it is, in fact, virtual, and I think leaving destructors and relying on implicitly virtual destructors seems like a needless foot gun in general in C++ OOP (since later refactoring could change the assumptions).